src/mito2/src/cache/write_cache.rs

```diff
@@ -65,11 +70,60 @@ impl WriteCache {
         request: SstUploadRequest,
         write_opts: &WriteOptions,
     ) -> Result<Option<SstInfo>> {
-        // TODO(yingwen): Write to the local store and then upload.
-        // Now we write to the remote and ignore local cache.
-        let mut writer =
-            ParquetWriter::new(request.upload_path, request.metadata, request.remote_store);
-        writer.write_all(request.source, write_opts).await
+        let _timer = WRITE_AND_UPLOAD_ELAPSED_TOTAL.start_timer();
+
+        let region_id = request.metadata.region_id;
+        let file_id = request.file_id;
+
+        let cache_path = self.file_cache.cache_file_path((region_id, file_id));
+        // Write to FileCache.
+        let mut writer = ParquetWriter::new(
+            cache_path.clone(),
+            request.metadata,
+            self.file_cache.local_store(),
+        );
+
+        let sst_info = writer.write_all(request.source, write_opts).await?;
+
+        // Upload sst file to remote object store.
+        let upload_path = request.upload_path.clone();
+        let reader = self
+            .file_cache
+            .local_store()
+            .reader(&cache_path)
+            .await
+            .context(error::OpenDalSnafu)?;
+
+        let mut writer = request
+            .remote_store
+            .writer_with(&upload_path)
+            .buffer(DEFAULT_WRITE_BUFFER_SIZE.as_bytes() as usize)
+            .await
+            .context(error::OpenDalSnafu)?;
+
+        let n = futures::io::copy(reader, &mut writer)
+            .await
+            .context(error::UploadSstSnafu { region_id, file_id })?;
+
+        UPLOAD_BYTES_TOTAL.inc_by(n);
```

Comment:
Prefer a more readable name.
```suggestion
        let bytes_written = futures::io::copy(reader, &mut writer)
            .await
            .context(error::UploadSstSnafu { region_id, file_id })?;

        UPLOAD_BYTES_TOTAL.inc_by(bytes_written);
```